### ISSUE TITLE
CI_JOB_TOKEN interferes with LAB_CORE_TOKEN

### ISSUE THREAD

**Issue by JohnVillalovos** [zaquestion/lab#767]
I am trying to use `lab` in the Gitlab CI but wanted to use a different token then the one found in `CI_JOB_TOKEN`.

So I set both `LAB_CORE_HOST` and `LAB_CORE_TOKEN`.

```
export LAB_CORE_HOST="https://gitlab.com"
export LAB_CORE_TOKEN="praritrocks!"
```

And attempted the following:

```
$ lab
Config saved to /root/.config/lab/lab.toml
<snip>
$ lab --debug mr list
2021/12/08 01:17:33 DEBUG: client.go:549: performing request
2021/12/08 01:17:33 DEBUG: 	method = GET
2021/12/08 01:17:34 ERROR: mr_list.go:67: GET https://gitlab.com/api/v4/projects/super/secret/project: 401 {message: 401 Unauthorized}
```

But when I did:
```
$ unset CI_JOB_TOKEN
$ lab mr list
!15 Draft: Testing lab
```

**Comment by bmeneg** [comment]
Yes, that indeed happens: when using lab within a CI the variables `CI_JOB_TOKEN`, `CI_PROJECT_URL`, `CI_PROJECT_PATH` and `GITLAB_USER_LOGIN` takes precedence over the LAB_* ones.

You can check our `.gitlab-ci.yml` file and see that we actually assign the real value to the LAB_* first, then assign the CI_* vars based on the LAB_* that we just set.

**Comment by JohnVillalovos** [comment]
> Yes, that indeed happens: when using lab within a CI the variables `CI_JOB_TOKEN`, `CI_PROJECT_URL`, `CI_PROJECT_PATH` and `GITLAB_USER_LOGIN` takes precedence over the LAB_* ones.

Okay. Well the the project README lists them in order and 1 is LAB_CORE_* which would seem to imply it has precedence over the others is how I read it.

**Comment by JohnVillalovos** [comment]
> You can check our `.gitlab-ci.yml` file and see that we actually assign the real value to the LAB_* first, then assign the CI_* vars based on the LAB_* that we just set.

Looking at https://github.com/zaquestion/lab/blob/e1100d31c32ecc9cdf9b469162dbbe4cc46eb9b5/.gitlab-ci.yml#L56-L58

Seems strange to me as someone unfamiliar with the project why `CI_JOB_TOKEN` gets set to `CI_PROJECT_URL` which was set to `LAB_CORE_HOST` which was set to `https://gitlab.com` if I am reading that correctly.

I'll be honest that confuses my little brain 😊

**Comment by JohnVillalovos** [comment]
Ah! Farther down it makes more sense:

https://github.com/zaquestion/lab/blob/e1100d31c32ecc9cdf9b469162dbbe4cc46eb9b5/.gitlab-ci.yml#L68-L78

**Comment by JohnVillalovos** [comment]
@bmeneg I created a PR with a documentation update #768 to close this issue.

**Comment by bmeneg** [comment]
> > You can check our `.gitlab-ci.yml` file and see that we actually assign the real value to the LAB_* first, then assign the CI_* vars based on the LAB_* that we just set.
> 
> Looking at
> 
> https://github.com/zaquestion/lab/blob/e1100d31c32ecc9cdf9b469162dbbe4cc46eb9b5/.gitlab-ci.yml#L56-L58
> 
> Seems strange to me as someone unfamiliar with the project why `CI_JOB_TOKEN` gets set to `CI_PROJECT_URL` which was set to `LAB_CORE_HOST` which was set to `https://gitlab.com` if I am reading that correctly.
> 
> I'll be honest that confuses my little brain blush

Ohh, it seems broken indeed! lol
But the exports solves it. But yeah, should be fixed I think. (and even removed if not being really used. I'll check it further).
